templates/zerver/help/add-a-custom-playground.md

```diff
@@ -0,0 +1,46 @@
+# Add a custom playground
+
+{!admin-only.md!}
+
+Playgrounds allow users to open the contents of a message code block in
+an external playground of their choice. This decision is offered as a pop up
+when hovering over any Zulip message containing a code block. Playgrounds
+can be configured specific to a programming language and the same language
+can be specified using the [syntax highlighting format](https://zulip.com/help/format-your-message-using-markdown#code)
+when composing the message using Markdown.
+
+### Add a custom playground
+
+{start_tabs}
+
+{settings_tab|playground-settings}
+
+1. Under **Add a new playground**, enter a **Name**, **Language** and
+**URL prefix**.
+
+1. Click **Add playground**.
+
+{end_tabs}
+
+## Walkthrough with an example
+
+Consider the following example.
+
+* Name: `Rust playground`
+* Language: `Rust`
+* URL prefix: `https://play.rust-lang.org/?edition=2018&code=`
+
+When composing a message `rust` can be mentioned as the syntax highlighting
+language in the code blocks.
+
+~~~
+``` rust
+fn main() {
+    // A hello world Rust program for demo purposes
+    println!("Hello World!");
+}
+```
+~~~
+
+The user would then get a on-hover option to open the above code in the playground
+they had previously configured.
```

Comment:
We should add a "Technical details" section that mentions:
* You can configure multiple playgrounds for a given language; if you do that, the user will get to choose which playground to open the code in using the name.
* The "Language" field is the human-readable Pygments language for the programming language, and some brief description of how the mapping works (or how to see them).
* That you can make up your own language and configure a playground for that, for something custom.